requests/utils.py

```diff
@@ -87,7 +83,22 @@ def super_len(o):
             # is actually a special file descriptor like stdin. In this
             # instance, we don't know what the length is, so set it to zero and
             # let requests chunk it instead.
-            current_position = total_length
+            if total_length is not None:
+                current_position = total_length
+
+        if hasattr(o, 'seek') and total_length is None:
```

Comment:
Ok cool, we can probably table this then, at least until someone starts addressing #3079.